traversal/expression/Operation.java

```diff
@@ -0,0 +1,101 @@
+
+
+package com.vaticle.typedb.core.traversal.expression;
+
+import com.vaticle.typedb.core.common.exception.TypeDBException;
+import com.vaticle.typeql.lang.common.TypeQLToken;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import static com.vaticle.typedb.core.common.exception.ErrorMessage.Expression.OPERATION_NOT_FOUND;
+import static com.vaticle.typedb.core.common.exception.ErrorMessage.Internal.ILLEGAL_STATE;
+import static com.vaticle.typedb.core.common.exception.ErrorMessage.Internal.UNIMPLEMENTED;
+
+public class Operation {
+    static Expression<?> instantiate(TypeQLToken.Expression.Operation op, Expression<?> first, Expression<?> second) {
+        switch (op) {
+            case PLUS:  return instantiatePlus(first, second);
+            case MINUS: return instantiateMinus(first, second);
+            case TIMES: return instantiateTimes(first, second);
+            case DIV:   return instantiateDiv(first, second);
+            case MOD:   return instantiateMod(first, second);
+            case POW:   return instantiatePow(first, second);
+
+            default:
+                throw TypeDBException.of(ILLEGAL_STATE);
+        }
+    }
+
+    private static Expression<?> instantiatePlus(Expression<?> first, Expression<?> second) {
+        if (first.isLong()) {
+            if (second.isLong()) return new ExpressionBase.Binary.Long<>(first.asLong(), second.asLong(), (a,b) -> a + b);
+            if (second.isDouble()) return new ExpressionBase.Binary.Double<>(first.asLong(), second.asDouble(), (a,b) -> a.doubleValue() + b);
```

Comment:
these are still better as `else-if` as a better signal these are mutually exclusive, same below

Reply:
in fact you can get the same effect with:
```
if {
  if
  else if
} else if {
  if
  else if
}

throw
```